local.go

```diff
@@ -128,11 +121,49 @@ func (r *LocalRepository) Get(curl *charm.URL) (charm.Charm, error) {
 	if curl.Revision == -1 && latest != nil {
 		return latest, nil
 	}
-	return nil, charmNotFound(curl, r.Path)
+	return nil, entityNotFound(curl, r.Path)
 }
 
 // GetBundle implements Interface.GetBundle.
 func (r *LocalRepository) GetBundle(curl *charm.URL) (charm.Bundle, error) {
-	// TODO frankban: implement this.
-	return nil, errgo.New("not implemented yet")
+	if err := r.checkUrlAndPath(curl); err != nil {
+		return nil, err
+	}
+	if curl.Series != "bundle" {
+		return nil, errgo.Newf("expected a bundle URL, got charm URL %q", curl)
+	}
+	// Note that the bundle does not inherently own a name different than the
+	// directory name. Neither the name nor the revision are included in the
+	// bundle metadata.
+	// TODO frankban: handle bundle revisions, totally ignored for now.
+	path := filepath.Join(r.Path, curl.Series, curl.Name)
+	info, err := os.Stat(path)
+	if err != nil {
+		return nil, entityNotFound(curl, r.Path)
+	}
+	// Do not support bundle archives for the time being. What archive name
+	// should we use? What's the use case for compressing bundles anyway?
+	if !info.IsDir() {
+		return nil, entityNotFound(curl, r.Path)
+	}
+	return charm.ReadBundleDir(path)
+}
+
+// checkUrlAndPath checks that the given URL represents a local entity and that
+// the repository path exists.
+func (r *LocalRepository) checkUrlAndPath(curl *charm.URL) error {
+	if curl.Schema != "local" {
+		return fmt.Errorf("local repository got URL with non-local schema: %q", curl)
```

Comment:
The charm schemas are confusing to many, especially new people. Could this return a message with the expected format.